logger/main.go

```diff
@@ -2,102 +2,106 @@ package main
 
 import (
 	"flag"
-	"fmt"
 	"log"
 	"os"
 	"os/signal"
 	"strconv"
-	"strings"
 	"syscall"
 	"time"
 
-	"github.com/coreos/go-etcd/etcd"
-	"github.com/deis/deis/logger/syslogd"
+	"github.com/deis/deis/logger/configurer"
+	"github.com/deis/deis/logger/publisher"
+	"github.com/deis/deis/logger/storage"
+	"github.com/deis/deis/logger/syslogish"
+	"github.com/deis/deis/logger/weblog"
 )
 
 var (
 	logAddr         string
+	logHost         string
 	logPort         int
-	drainURI        string
 	enablePublish   bool
-	publishHost     string
-	publishPath     string
-	publishPort     string
+	webAddr         string
+	webPort         int
+	etcdHost        string
+	etcdPort        string
+	etcdPath        string
+	configInterval  int
 	publishInterval int
 	publishTTL      int
 )
 
 func init() {
 	flag.StringVar(&logAddr, "log-addr", "0.0.0.0", "bind address for the logger")
+	flag.StringVar(&logHost, "log-host", getopt("HOST", "127.0.0.1"),
+		"address of the host running logger")
 	flag.IntVar(&logPort, "log-port", 514, "bind port for the logger")
-	flag.StringVar(&drainURI, "drain-uri", "", "default drainURI, once set in etcd, this has no effect.")
-	flag.StringVar(&syslogd.LogRoot, "log-root", "/data/logs", "log path to store logs")
+	flag.StringVar(&configurer.DefaultDrainURI, "drain-uri", "",
+		"default drainURI, once set in etcd, this has no effect.")
+	flag.StringVar(&storage.LogRoot, "log-root", "/data/logs", "log path to store logs")
+	flag.StringVar(&webAddr, "web-addr", "0.0.0.0", "bind address for the web service")
+	flag.IntVar(&webPort, "web-port", 8088, "bind port for the web service")
 	flag.BoolVar(&enablePublish, "enable-publish", false, "enable publishing to service discovery")
-	flag.StringVar(&publishHost, "publish-host", getopt("HOST", "127.0.0.1"), "service discovery hostname")
+	// Support legacy flag names even though the variable names have been refactored
+	flag.StringVar(&etcdHost, "publish-host", getopt("HOST", "127.0.0.1"),
+		"service discovery hostname")
+	flag.StringVar(&etcdPort, "publish-port", getopt("ETCD_PORT", "4001"), "service discovery port")
+	flag.StringVar(&etcdPath, "publish-path", getopt("ETCD_PATH", "/deis/logs"),
+		"path to publish host/port information")
+	flag.IntVar(&configInterval, "config-interval", 10, "config interval in seconds")
 	flag.IntVar(&publishInterval, "publish-interval", 10, "publish interval in seconds")
-	flag.StringVar(&publishPath, "publish-path", getopt("ETCD_PATH", "/deis/logs"), "path to publish host/port information")
-	flag.StringVar(&publishPort, "publish-port", getopt("ETCD_PORT", "4001"), "service discovery port")
 	flag.IntVar(&publishTTL, "publish-ttl", publishInterval*2, "publish TTL in seconds")
```

Comment:
I do know that you didn't change but since it's a rewrite you might as well. Basically when you define flags as (just one flag here but you get the idea)

```
var (
    publishPath = flag.StringVar(&publishPath, "publish-path", getopt("ETCD_PATH", "/deis/logs"), "path to publish host/port information")
)
```

you can access the value using `*publishPath` which is less pretty but the compiler already does the work. When you do it the way it's done now (through `init`) the binary sets up the memory on boot. It's not much of a performance improvement but it's something.

I leave it to you to decide.